### PR TITLE
feat: rewrite MC2SmearedParticle to operate on collections

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This reworks the MC2SmearedParticle algorithm and factory to operate on podio collections, which are passed to the algorithm. It moves the loop over MCParticles into the algorithm (maybe at some point the factories can parallelize this, but we're not there, and consistency in how algorithms interact with collections is more important right now). It also removes the (defunct) smearing configuration.

This also removes the `generatorStatus == 0` entries as done in #772.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue: use collections in MC2SmearedParticles)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No.

### CODE REVIEW

**Comment by wdconinc** [comment]
Include #772 too.

File: src/algorithms/reco/MC2SmearedParticle.cc [hunk 1]
```diff
@@ -2,61 +2,46 @@
...
-    if (mc_particle->getGeneratorStatus() > 1) {
-        m_log->debug("ignoring particle with generatorStatus = {}", mc_particle->getGeneratorStatus());
-        return nullptr;
+std::unique_ptr<edm4eic::ReconstructedParticleCollection> eicrecon::MC2SmearedParticle::produce(const edm4hep::MCParticleCollection* mc_particles) {
+    auto rec_particles = std::make_unique<edm4eic::ReconstructedParticleCollection>();
+    for (const auto& mc_particle: *mc_particles) {
+
+        if (mc_particle.getGeneratorStatus() != 1) {
+            m_log->debug("ignoring particle with generatorStatus = {}", mc_particle.getGeneratorStatus());
+            return nullptr;
```
Comment (wdconinc) [inline src/algorithms/reco/MC2SmearedParticle.cc:21]:
```suggestion
            continue;
```

**Comment by wdconinc** [comment]
This differs from the current `main` branch in the removal of non-generated `GeneratedParticles` only (sometimes O(1000) per event for large s, large Q<sup>2</sup> events), and this does not differ at all from the artifacts in #772 (in particular https://github.com/eic/EICrecon/actions/runs/5613960299).

**Review by veprbl — DISMISSED** [review]
LGTM

**Review by veprbl — APPROVED** [review]

**Pull request merged**